aptos-move/aptos-vm-types/src/output.rs

```diff
@@ -74,12 +74,17 @@ impl VMOutput {
         // First, check if output of transaction should be discarded or delta
         // change set is empty. In both cases, we do not need to apply any
         // deltas and can return immediately.
-        if self.status().is_discarded() || self.change_set().aggregator_v1_delta_set().is_empty() {
+        if self.status().is_discarded()
+            || (self.change_set().aggregator_v1_delta_set().is_empty()
+                && self.change_set().aggregator_v2_change_set().is_empty())
+        {
             return Ok(self);
         }
 
         let (change_set, fee_statement, status) = self.unpack_with_fee_statement();
-        let materialized_change_set = change_set.try_materialize(state_view)?;
+        let materialized_change_set = change_set
+            .try_materialize_aggregator_v1_delta_set(state_view)?
+            .try_materialize_aggregator_v2_changes(state_view)?;
```

Comment:
when we return the actual output from block executor, we need all V2 changes cleared and the write outputs to be updated / replaced properly, at least that's the current expectation in the code. this isn't accomplished atm, so let's delete the try_materialize_v2_changes and revisit /address after this PR lands (as a part of materialization effort)